macaroonbakery/bakery.py

```diff
@@ -1,171 +1,18 @@
-        @return a discharge macaroon.
-        @raise DischargeError when an error occurs during the discharge
-            process.
-        '''
-        headers = DEFAULT_PROTOCOL_VERSION
-        payload = {'id': third_party_caveat_condition}
-
-        response = requests.post(third_party_location + '/discharge',
-                                 headers=headers,
-                                 data=payload,
-                                 # timeout=TIME_OUT, TODO: add a time out
-                                 cookies=self._jar)
-        status_code = response.status_code
-        if status_code == 200:
-            return _extract_macaroon_from_response(response)
-        if (status_code == 401 and
-                response.headers.get('WWW-Authenticate') == 'Macaroon'):
-            error = response.json()
-            if error.get('Code', '') != ERR_INTERACTION_REQUIRED:
-                return DischargeException('unable to get code from discharge')
-            info = _extract_urls(response)
-            self._visit_page(info.visit_url)
-            # Wait on the wait url and then get a macaroon if validated.
-            return _acquire_macaroon_from_wait(info.wait_url)
-
-
-def _extract_macaroon_from_response(response):
-    '''Extract the macaroon from a direct successful discharge.
-
-    @param response from direct successful discharge.
-    @return a macaroon object.
-    @raises DischargeError if any error happens.
-    '''
-    response_json = response.json()
-    return utils.deserialize(response_json['Macaroon'])
-
-
-def _acquire_macaroon_from_wait(wait_url):
-    ''' Return the macaroon acquired from the wait endpoint.
-
-    Note that will block until the user interaction has completed.
-
-    @param wait_url the get url to call to get a macaroon.
-    @return a macaroon object
-    @raises DischargeError if any error happens.
-    '''
-    resp = requests.get(wait_url, headers=DEFAULT_PROTOCOL_VERSION)
-    response_json = resp.json()
-    macaroon = response_json['Macaroon']
-    return utils.deserialize(macaroon)
-
-
-def _extract_urls(response):
-    '''Return _Info of the visit and wait URL from response.
-
-    @param response the response from the discharge endpoint.
-    @return a _Info object of the visit and wait URL.
-    @raises DischargeError for ant error during the process response.
-    '''
-    response_json = response.json()
-    visit_url = response_json['Info']['VisitURL']
-    wait_url = response_json['Info']['WaitURL']
-    return _Info(visit_url=visit_url, wait_url=wait_url)
+from macaroonbakery.authorizer import ClosedAuthorizer
 
 
 class Bakery(object):
     '''Convenience class that contains both an Oven and a Checker.
     '''
     def __init__(self, location=None, locator=None, ops_store=None, key=None,
                  identity_client=None, checker=None, root_key_store=None,
-                 authorizer=None):
+                 authorizer=ClosedAuthorizer()):
```

Comment:
A better idiom (more pythonic) would be to keep the default as None but initialize it to an instance inside the init method, eg

if authorizer is None:
    authorizer = ClosedAuthorizer()

Reply:
As discussed online, given that ClosedAuthorizer is immutable, I think it's OK as is (and slightly more understandable).